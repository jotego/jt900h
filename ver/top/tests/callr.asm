    ; calr
    maxmode on
    relaxed on
    org 0
    ld a,0xbf    ; common header

    ld xsp, stack
    calr check1
check2:
    jp bad_end
check1:
    cp xsp,stack-4
    jp ne,bad_end

    ld xwa,(xsp)
    cp xwa,check2
    jp ne, bad_end

end_loop:
    ld hl,0xbabe
    ld (0xffff),0xff
    jp end_loop
bad_end:
    ld hl,0xdead
    jp bad_end
data:
    dw 0xcafe,0xbeef,0xffff,0xeeee,0xcccc
stack:
    dw 0,0,0,0,0,0,0,0,0,0
    end