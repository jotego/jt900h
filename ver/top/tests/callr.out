0000000a - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
0000babe - 00000000 - 00000000 - 00000000
-- index registers --
00000000 - 00000000 - 00000000 - 0000003f
SR = 42f8
